### PR TITLE
Bug 1818278 - Update firefox-android documentations around Beta cuts

### PR DESCRIPTION
**Pull request by gabrielluong**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1818278

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -4,13 +4,13 @@ title: Changelog
 permalink: /changelog/
 ---
 
-# 115.0.0 (In Development)
+# 115.0 (In Development)
```
Comment (gabrielluong) [inline docs/changelog.md:7]: I noticed starting on 110.0 that we basically dropped the extra `.0` to the versioning https://maven.mozilla.org/?prefix=maven2/org/mozilla/components/compose-tabstray/

File: docs/contribute/release_checklist.md [hunk 2]
```diff
@@ -11,9 +11,51 @@ These are instructions for preparing a release branch for Firefox Android and st
 **This part is 100% covered by the Release Management team. The dev team should not perform these steps.**
 
 1. Create a branch name with the format `releases_v[beta_version]` off of the `main` branch (for example, `releases_v109`) through the GitHub UI.
-`[beta_version]` should follow the Firefox Beta release number. See [Firefox Release Calendar](https://wiki.mozilla.org/Release_Management/Calendar).
-2. In `main` [version.txt](https://github.com/mozilla-mobile/firefox-android/blob/main/version.txt):
-   - Update the version from `[previous_nightly_version].0a1` to `[nightly_version].0a1`. For example:
+`[beta_version]` should follow the Firefox Beta release number. See [Firefox Release Calendar](https://wiki.mozilla.org/Release_Management/Calendar). The following below should be performed in this new beta release branch:
```
Comment (DonalMe) [inline docs/contribute/release_checklist.md:14]: The wiki page is being deprecated - replace the URL with https://whattrainisitnow.com/calendar/

File: docs/contribute/release_checklist.md [hunk 3]
```diff
@@ -22,110 +64,73 @@ These are instructions for preparing a release branch for Firefox Android and st
...
--// These lines are generated by android-components/automation/application-services-nightly-bump.py
--val VERSION = "114.20230420185101"
--val CHANNEL = ApplicationServicesChannel.NIGHTLY
-+val VERSION = "114.0"
-+val CHANNEL = ApplicationServicesChannel.RELEASE
-```
-10. Create a commit named `Swift to Application Services Release`.  (Note: application-services releases directly after the nightly cycle, there's no beta cycle).
-11. Once all three of the above commits have landed, create a new `Firefox Android (Android-Components, Fenix, Focus)` release in [Ship-It](https://shipit.mozilla-releng.net/) and continue with the release checklist per normal practice.
+4. Once all three of the above commits have landed, create a new `Firefox Android (Android-Components, Fenix, Focus)` release in [Ship-It](https://shipit.mozilla-releng.net/) and continue with the release checklist per normal practice.
+5. The Github Action will automatically bump the GeckoView version once the new Beta build is created and uploaded to Maven, as shown in [#324](https://github.com/mozilla-mobile/firefox-android/pull/324) for example.
```
Comment (DonalMe) [inline docs/contribute/release_checklist.md:69]: Swap the content in Step 4 and  Step 5
The final step is setting up the build in ShipIt after the automatic GeckoView version bump PR is merged.
"Once all four of the above commits have landed..."

**Review by DonalMe — APPROVED** [review]
LGTM

File: docs/contribute/release_checklist.md [hunk 4]
```diff
@@ -11,9 +11,51 @@ These are instructions for preparing a release branch for Firefox Android and st
...
+    - In [version.txt](https://github.com/mozilla-mobile/firefox-android/blob/main/version.txt), update the version from `[beta_version].0a1` to `[beta_version].0b1`. Create a commit named `Set version to [beta_version].0b1` for this change. This change can either be directly committed to the `releases_v[beta_version]` branch or a pull request can be created against it and then merged.
+        ```diff
+        diff --git a/version.txt b/version.txt
+        --- a/version.txt
+        +++ b/version.txt
+        @@ -1 +1 @@
+        -109.0a1
+        +109.0b1
+        ```
+    - In [Gecko.kt](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/plugins/dependencies/src/main/java/Gecko.kt), set the `channel` to `GeckoChannel.BETA`. Create a commit named `Switch to GeckoView Beta` for this change. This change can either be directly committed to the `releases_v[beta_version]` branch or a pull request can be created against it and then merged. Once landed, it is expected that this change will temporarily break builds on the branch. The next step will fix them.
```
Comment (rvandermeulen) [inline docs/contribute/release_checklist.md:24]: Please change `The next step will fix them.` to `Step #4 will fix them.`

File: docs/contribute/release_checklist.md [hunk 5]
```diff
@@ -22,110 +64,73 @@ These are instructions for preparing a release branch for Firefox Android and st
...
--// These lines are generated by android-components/automation/application-services-nightly-bump.py
--val VERSION = "114.20230420185101"
--val CHANNEL = ApplicationServicesChannel.NIGHTLY
-+val VERSION = "114.0"
-+val CHANNEL = ApplicationServicesChannel.RELEASE
-```
-10. Create a commit named `Swift to Application Services Release`.  (Note: application-services releases directly after the nightly cycle, there's no beta cycle).
-11. Once all three of the above commits have landed, create a new `Firefox Android (Android-Components, Fenix, Focus)` release in [Ship-It](https://shipit.mozilla-releng.net/) and continue with the release checklist per normal practice.
+4. The Github Action will automatically bump the GeckoView version once the new Beta build is created and uploaded to Maven, as shown in [#324](https://github.com/mozilla-mobile/firefox-android/pull/324) for example.
+5. Once all four of the above commits have landed, create a new `Firefox Android (Android-Components, Fenix, Focus)` release in [Ship-It](https://shipit.mozilla-releng.net/) and continue with the release checklist per normal practice.
```
Comment (rvandermeulen) [inline docs/contribute/release_checklist.md:69]: Please remove the `four` from this.

**Review by rvandermeulen — APPROVED** [review]
LGTM, just a couple very minor nits. Thanks very much for doing this! It's a big improvement.

**Pull request merged**